apps/apps.c

```diff
@@ -2294,29 +2294,28 @@ int app_access(const char* name, int flag)
 #ifdef _WIN32
 int app_isdir(const char *name)
 {
-    HANDLE hList;
-    WIN32_FIND_DATA FileData;
+    DWORD attr;
 # if defined(UNICODE) || defined(_UNICODE)
     size_t i, len_0 = strlen(name) + 1;
+    WCHAR tempname[MAX_PATH];
 
-    if (len_0 > OSSL_NELEM(FileData.cFileName))
+    if (len_0 > MAX_PATH)
         return -1;
 
 #  if !defined(_WIN32_WCE) || _WIN32_WCE>=101
     if (!MultiByteToWideChar
-        (CP_ACP, 0, name, len_0, FileData.cFileName, len_0))
+        (CP_ACP, 0, name, len_0, tempname, len_0))
```

Comment:
I think it's guaranteed that number of WCHARs is less than or equal to number of chars. If one Unicode character requires two WCHARs (i.e. 4 bytes) in UTF-16, its code point is at least 0x10000. I don't think there's any encoding scheme using one byte to represent such a character.

Or do you want me to change the second `len_0` to `MAX_PATH`?